pkgs/applications/graphics/rawtherapee/default.nix

```diff
@@ -1,29 +1,34 @@
-{ stdenv, fetchurl, pkgconfig, gtk, cmake, pixman, libpthreadstubs, gtkmm, libXau
-, libXdmcp, lcms2, libiptcdata, libcanberra, fftw, expat
+{ stdenv, fetchFromGitHub, pkgconfig, gtk, cmake, pixman, libpthreadstubs, gtkmm, libXau
+, libXdmcp, lcms2, libiptcdata, libcanberra, fftw, expat, pcre, libsigcxx 
 , mercurial  # Not really needed for anything, but it fails if it does not find 'hg'
 }:
 
 stdenv.mkDerivation rec {
-  name = "rawtherapee-4.0.10";
+  name = "rawtherapee-4.2";
   
-  src = fetchurl {
-    url = http://rawtherapee.googlecode.com/files/rawtherapee-4.0.10.tar.xz;
-    sha256 = "1ibsdm2kqpw796rcdihnnp67vx0wm1d1bnlzq269r9p01w5s102g";
+  src = fetchFromGitHub {
+    owner = "Beep6581";
+    repo = "RawTherapee";
+    rev = "4.2";
+    sha256 = "1v4px239vlmk9l8wbzlvlyni4ns12icxmgfz21m86jkd10pj5dgr";
   };
   
   buildInputs = [ pkgconfig gtk cmake pixman libpthreadstubs gtkmm libXau libXdmcp
-    lcms2 libiptcdata mercurial libcanberra fftw expat ];
+    lcms2 libiptcdata mercurial libcanberra fftw expat pcre libsigcxx ];
 
-  # Disable the use of the RAWZOR propietary libraries
-  cmakeFlags = [ "-DWITH_RAWZOR=OFF" ];
+  patchPhase = ''
```

Comment:
I believe adding the patch to `patches` is sufficient, then the standard patch phase takes care of it.